whenchanged/whenchanged.py

```diff
@@ -121,33 +122,30 @@ def is_interested(self, path):
 
         return False
 
-    def on_change(self, path):
+    def on_change(self, path, event = None):
+        if event is not None and not event.is_directory:
+            self.last_event_type = event.event_type
+            self.set_envvar('event', 'file_' + event.event_type)
         if self.is_interested(path):
             self.run_command(path)
 
     def on_created(self, event):
-        if self.observer.__class__.__name__ == 'InotifyObserver':
-            # inotify also generates modified events for created files
-            return
-
         if not event.is_directory:
-            self.set_envvar('event', 'file_created')
-            self.on_change(event.src_path)
+            self.on_change(event.src_path, event)
 
     def on_modified(self, event):
-        if not event.is_directory:
-            self.set_envvar('event', 'file_modified')
-            self.on_change(event.src_path)
+        if not event.is_directory and self.last_event_type != 'created':
```

Comment:
Is it sufficient to test only for the last event type, regardless of which file triggered the event?

Reply:
I think, but do you refer to some situation in particular?

Reply:
I'm not sure if it's possible, but there may be a race condition in case:
file1 is created
inotify fires created event for file1
-> on_change for file1
-> last_event_type = created
file2 is modified
inotify fires modified event for file2
-> ignored because last_event_type == created
-> last_event_type = modified
inotify fires modified event for file1
-> on_change for file1 again